osu.Game.Tests/Visual/Gameplay/TestSceneHUDOverlay.cs

```diff
@@ -149,6 +149,41 @@ public void TestChangeHUDVisibilityOnHiddenKeyCounter()
             AddAssert("key counters still hidden", () => !keyCounterFlow.IsPresent);
         }
 
+        [Test]
+        public void TestHoldForMenuDoesWorkWhenHidden()
+        {
+            bool activated = false;
+
+            HoldForMenuButton getHoldForMenu() => hudOverlay.ChildrenOfType<HoldForMenuButton>().Single();
+
+            createNew();
+
+            AddStep("bind action", () =>
+            {
+                activated = false;
+
+                var holdForMenu = getHoldForMenu();
+
+                holdForMenu.Action += () => activated = true;
+            });
+
+            AddStep("set showhud false", () => hudOverlay.ShowHud.Value = false);
+            AddUntilStep("hidetarget is hidden", () => !hideTarget.IsPresent);
+
+            AddStep("attempt activate", () =>
+            {
+                InputManager.MoveMouseTo(getHoldForMenu().OfType<HoldToConfirmContainer>().Single());
```

Comment:
Threw me off for a hot minute that this was enumerating the children of the `HoldForMenuButton` but sure.